index/scorch/scorch.go

```diff
@@ -407,16 +407,20 @@ func (s *Scorch) AddEligibleForRemoval(epoch uint64) {
 	s.rootLock.Unlock()
 }
 
-func (s *Scorch) MemoryUsed() uint64 {
+func (s *Scorch) CurDirtyBytes() uint64 {
 	var memUsed uint64
 	s.rootLock.RLock()
-	for _, segmentSnapshot := range s.root.segment {
-		memUsed += 8 /* size of id -> uint64 */ +
-			segmentSnapshot.segment.SizeInBytes()
-		if segmentSnapshot.deleted != nil {
-			memUsed += segmentSnapshot.deleted.GetSizeInBytes()
+	if s.root != nil {
+		for _, segmentSnapshot := range s.root.segment {
+			memUsed += 8 /* size of id -> uint64 */
+			if _, ok := segmentSnapshot.segment.(*mem.Segment); ok {
```

Comment:
sorry, I am not able to follow the logic here,  why are we restricting the memory used computations only to mem segments? At any point in time , there could be a mix of both mem and zap segments within an index snapshot right? 
I could see in the earlier logic, we were calling the segment.SizeInBytes() API which was implemented by both mem and zap segments..

Reply:
Whoops...Nevermind, I read the commit msg now.. :)